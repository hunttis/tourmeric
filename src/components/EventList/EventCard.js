import React, { Component, Fragment } from 'react';
import { Translate } from 'react-localize-redux';
import PropTypes from 'prop-types';
import Moment from 'react-moment';
import moment from 'moment';
import _ from 'lodash';

import { ParticipateButton } from './ParticipateButton';
import { participantCount, checkParticipation } from '../../api/eventApi';
import EditableField from '../Common/EditableField-container';
import EventModal from './EventModal-container';

export default class EventCard extends Component {

  state = { modalOpen: false };

  openModal() {
    this.setState({ modalOpen: true });
  }

  closeModal() {
    this.setState({ modalOpen: false });
  }

  render() {
    const { eventId, events, userid, profile, participations, settings, categories } = this.props;

    const dateFormat = _.get(settings, 'dateFormat', 'DD-MM-YYYY');
    const dateFormatWithDayName = `${dateFormat} (dd)`;
    const eventContent = _.find(events, ['key', eventId]).value;
    const category = categories[eventContent.category];
    const alreadyParticipated = checkParticipation(userid, eventId, participations);
    const thisParticipation = _.get(participations, `${eventId}.${userid}`, []);
    const maxParticipants = _.get(eventContent, 'playerSlots', 0);
    const currentParticipants = participantCount(eventId, participations);
    const eventFull = maxParticipants <= currentParticipants;

    return (
      <Fragment>
        {this.state.modalOpen &&
          <EventModal key={`modal${eventId}`} eventId={eventId} closeModal={() => this.closeModal()} />
        }
        <div className="column is-12 eventcard">
          <h2 className="subtitle date-item">
            <Moment format={dateFormatWithDayName}>{eventContent.date}</Moment>
          </h2>
          <div className="card card-shadow">

            <div className="card-header eventheader">
              <div className="card-header-icon eventheader-text tooltip" data-tooltip={category.name}>
                <img className="image is-64x64" src={category.image} alt="" />
                {category.abbreviation}:
              </div>
              <div className="card-header-title eventheader-text">
                <div>{eventContent.name}</div>
              </div>
            </div>

            <div className="card-content">
              <div className="has-icons-left card-item">
                <span className="tooltip" data-tooltip="Starting time">
                  <i className="fas fa-clock" />&nbsp;&nbsp;{eventContent.time}
                </span>
              </div>
              <div className="has-icons-left card-item">
                <span className="tooltip" data-tooltip="Entry fee">
                  <i className="fas fa-money-bill-alt" />&nbsp;&nbsp;{eventContent.entryFee}&nbsp;â‚¬
                </span>
              </div>

              <div className={`has-icons-left card-item ${eventFull && 'has-text-warning'}`}>
                <span className="tooltip" data-tooltip="People already enrolled">
                  <i className="fas fa-users" />&nbsp;&nbsp;{currentParticipants}&nbsp;
                  {eventContent.playerSlots &&
                  <span>/&nbsp;{eventContent.playerSlots}&nbsp;
                    {eventFull && <span>(<Translate id="eventfull" />)</span>}
                  </span>}
                </span>
              </div>

              <div className="has-icons-left card-item">
                <div className="level">
                  {eventContent.format &&
                    <div className="level-left card-footer-items">
                      <span className="tooltip card-margin-right" data-tooltip="Format">
                        <i className="fas fa-book" />&nbsp;&nbsp;{eventContent.format}&nbsp;
                      </span>
                    </div>
                  }

                  {eventContent.rules &&
                    <div className="level-item card-footer-items">
                      <span className="tooltip card-margin-right" data-tooltip="Rules Level">
                        <i className="fas fa-balance-scale" />&nbsp;&nbsp;{eventContent.rulesLevel}
                      </span>
                    </div>
                  }

                  <div className="level-right card-footer-items ">
                    <a onClick={() => this.openModal()} className="card-footer-link">
                      <i className="fas fa-trophy" />&nbsp;&nbsp;<Translate id="allinfo" />
                    </a>
                  </div>

                </div>
              </div>

            </div>

            {/* CARD FOOTER ON TABLET AND ABOVE */}
            <div className="card-footer is-hidden-mobile">
              { alreadyParticipated &&
              <div className="card-footer-item event-card-footer">
                <EditableField
                  inputClasses="is-rounded"
                  leftIcon="comment"
                  labelContent=""
                  placeHolder="comment"
                  defaultValue={thisParticipation.comment}
                  path={`/participations/${eventId}/${userid}`}
                  targetName="comment"
                />
              </div>
              }
              <div className="card-footer-item event-card-footer">
                <ParticipateButton userId={userid} profile={profile} eventId={eventId} participations={participations} waitList={eventFull} />
              </div>
            </div>
            {/* END CARD FOOTER */}

            {/* CARD FOOTER ON MOBILE */}
            { alreadyParticipated &&
            <Fragment>
              <div className="card-footer is-hidden-tablet">
                <div className="card-footer-item event-card-footer">
                  <EditableField
                    inputClasses="is-rounded"
                    leftIcon="comment"
                    labelContent=""
                    placeHolder="comment"
                    defaultValue={thisParticipation.comment}
                    path={`/participations/${eventId}/${userid}`}
                    targetName="comment"
                  />
                </div>
              </div>
            </Fragment>
            }
            <div className="card-footer is-hidden-tablet">
              <div className="card-footer-item event-card-footer">
                <ParticipateButton userId={userid} profile={profile} eventId={eventId} participations={participations} waitList={eventFull} />
              </div>
            </div>
            {/* END CARD FOOTER */}

          </div>
          <p>&nbsp;</p>
          { moment(eventContent.date).isSameOrAfter(moment(), 'day') &&
          <div className="level participation-level">

            { !alreadyParticipated &&
              <div className="level-left is-hidden-mobile" />
            }

          </div>
        }
        </div>
      </Fragment>
    );
  }
}

EventCard.propTypes = {
  eventId: PropTypes.string,
  userid: PropTypes.string,
  profile: PropTypes.object,
  events: PropTypes.array,
  settings: PropTypes.object,
  participations: PropTypes.object,
  categories: PropTypes.object,
};
