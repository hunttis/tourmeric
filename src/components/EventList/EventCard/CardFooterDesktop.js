import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import EditableField from '../../Common/EditableField-container';
import ParticipateButton from '../ParticipateButton-container';

export const CardFooterDesktop = ({ alreadyParticipated, thisParticipation, eventId, userId }) => (
  <Fragment>
    <div className="card-footer is-hidden-mobile">
      { alreadyParticipated &&
        <div className="card-footer-item event-card-footer">
          <EditableField
            inputClasses="is-rounded"
            leftIcon="comment"
            labelContent=""
            placeHolder="comment"
            defaultValue={thisParticipation.comment}
            path={`/participations/${eventId}/${userId}`}
            targetName="comment"
          />
        </div>
      }
      <div className="card-footer-item event-card-footer">
        <ParticipateButton eventId={eventId} />
      </div>
    </div>
  </Fragment>
);

CardFooterDesktop.propTypes = {
  alreadyParticipated: PropTypes.bool.isRequired,
  thisParticipation: PropTypes.object.isRequired,
  eventId: PropTypes.string.isRequired,
  userId: PropTypes.string,
};
