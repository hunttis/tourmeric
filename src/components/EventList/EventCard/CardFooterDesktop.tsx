import React from 'react';

import EditableField from '../../Common/EditableField-container';
import ParticipateButton from '../ParticipateButton-container';
import { Participation } from '../../../models/ReduxState';

interface Props {
  alreadyParticipated: boolean;
  thisParticipation: Participation;
  eventId: string;
  userId: string;
}

export const CardFooterDesktop = ({ alreadyParticipated, thisParticipation, eventId, userId }: Props) => (
  <>
    <div className="card-footer is-hidden-mobile">

      {alreadyParticipated &&
        <div className="card-footer-item event-card-footer">
          <EditableField
            inputClasses="is-rounded"
            leftIcon="comment"
            labelContent=""
            placeHolder="comment"
            defaultValue={thisParticipation.comment}
            path={`/participations/${eventId}/${userId}`}
            targetName="comment"
          />
        </div>
      }
      <div className="card-footer-item event-card-footer">
        <ParticipateButton eventId={eventId} />
      </div>
    </div>
  </>
);
