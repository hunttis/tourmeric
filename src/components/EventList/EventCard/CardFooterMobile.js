import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import EditableField from '../../Common/EditableField-container';
import ParticipateButton from '../ParticipateButton-container';

export const CardFooterMobile = ({ alreadyParticipated, thisParticipation, eventId, userId }) => (
  <Fragment>
    { alreadyParticipated &&
    <div className="card-footer is-hidden-tablet">
      <div className="card-footer-item event-card-footer">
        <EditableField
          inputClasses="is-rounded"
          leftIcon="comment"
          labelContent=""
          placeHolder="comment"
          defaultValue={thisParticipation.comment}
          path={`/participations/${eventId}/${userId}`}
          targetName="comment"
        />
      </div>
    </div>
    }
    <div className="card-footer is-hidden-tablet">
      <div className="card-footer-item event-card-footer">
        <ParticipateButton eventId={eventId} />
      </div>
    </div>
  </Fragment>
);

CardFooterMobile.propTypes = {
  alreadyParticipated: PropTypes.bool.isRequired,
  thisParticipation: PropTypes.object.isRequired,
  eventId: PropTypes.string.isRequired,
  userId: PropTypes.string,
};
