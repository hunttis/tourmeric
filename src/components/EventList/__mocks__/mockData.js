export const mockEventId = 'mockEventId1';

export const mockUserId = 'mockUserId';

export const mockProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{ displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890' }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
};

export const mockUnorderedEvents = {
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents = [
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png' },
  };

export const mockParticipations = { mockEventId1: { mockUserId: { comment: 'Jee',
  date: '2018-06-23T09:40:26.992Z',
  firstName: 'Test',
  lastName: 'Person',
  userId: 'mockUserId' } } };

export const mockCategories = { category1: { abbreviation: 'WH40K',
  logo: '1527227370219137123',
  name: 'Warhammer 40,000',
  type: 'Miniatures' },
category2: { abbreviation: 'MtG',
  logo: '152722965771313789',
  name: 'Magic: The Gathering',
  type: 'TCG' },
'-LAlvCmIWNzYKpa7-7HI': { abbreviation: 'Lautapelit',
  logo: '152719272536668394',
  name: 'Lautapelit / Board games' },
'-LCmESny8m9JVEFQ_qOt': { abbreviation: 'PKM',
  logo: '152722747904159438',
  name: 'Pok√©mon',
  type: 'TCG' },
'-LDImTAGxkzrorL0tQdi': { abbreviation: 'VG',
  logo: '152719372155055672',
  name: 'Videogames' },
'-LDIw-1LFgiM0EJZGfus': { abbreviation: 'Necromunda',
  logo: '15271953483568546',
  name: 'Necromunda',
  type: 'Miniatures' } };
