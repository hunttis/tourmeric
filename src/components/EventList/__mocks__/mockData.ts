import { FirebaseProfile, Participation, User, ProviderData, Location } from '~/models/ReduxState';
import { TourmericEvent } from '~/models/Events';
import { Settings } from '~/models/Settings';
import { UploadedFile, Category } from '~/models/Category';

export const mockEventId = 'mockEvent1';

export const mockUserId = 'mockUserId';

export const mockLocation: Location = {
  pathname: '/events/2018/10',
  search: 'search',
  hash: 'hash',
  key: 'key',
};

export const mockProviderData: ProviderData = {
  displayName: 'MockName',
  email: 'mock@userton.com',
  photoURL: 'http://url',
  providerId: 'facebook',
  uid: '1234567890',
};

export const mockUser: User = {
  avatarUrl: 'http://www.google.fi/google.png',
  displayName: 'Mock Userton',
  firstName: 'Mock',
  lastName: 'Userton',
  providerData: mockProviderData,
  email: 'mock@userton.com',
  username: 'Mockie',
  dciNumber: '123456',
  role: 'user',
  active: true,
};

export const mockProfile: FirebaseProfile = {
  avatarUrl: 'https://foo.com/picture',
  dciNumber: '123512351112',
  displayName: 'Test Person',
  email: 'fake@fake.com',
  firstName: 'Test',
  lastName: 'Person',
  providerData: [{
    displayName: 'Test Person',
    email: 'fake@fake.com',
    photoURL: 'https://foo.com/picture',
    providerId: 'facebook.com',
    uid: '1234567890',
  }],
  role: 'admin',
  isEmpty: false,
  isLoaded: true,
};

export const mockSingleEvent: TourmericEvent = {
  category: 'category1',
  createDate: '2018-05-24T17:00:46.500Z',
  date: '2018-09-24',
  entryFee: '10',
  format: 'Miniatures',
  link: 'http://www.google.fi',
  name: 'Warhammer thing',
  notes: 'Bring your war face!',
  playerSlots: '10',
  prizes: 'Glory!',
  published: true,
  rulesLevel: 'Casual',
  time: 'Kill everything',
  eventType: 'single',
};

export const mockUnorderedEvents: {[key: string]: TourmericEvent} = {
  mockEvent1: mockSingleEvent,
  mockEvent2: {
    category: 'category2',
    createDate: '2018-05-20T16:46:32.009Z',
    date: '2018-05-20',
    entryFee: '6',
    format: 'Standard',
    name: 'Tapahtuma',
    notes: 'FIJoijgasdsdgasdggagads',
    playerSlots: '14',
    prizes: 'Foooooooooooooooo',
    rulesLevel: 'Competitive',
    time: '10:00',
  },
  mockEvent3: {
    category: 'category2',
    createDate: '2018-05-20T16:04:29.595Z',
    date: '2018-05-21',
    entryFee: '5',
    format: 'Modern',
    link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
    name: 'Monday Modern',
    notes: 'Bring a friend!',
    playerSlots: '31',
    prizes: 'Three (3) OMG credits per player into the prizepool.',
    published: true,
    rulesLevel: 'Regular',
    time: '19:00',
  },
};

export const mockEvents: {key: string, value: TourmericEvent}[] = [
  { key: 'mockEvent1',
    value: mockSingleEvent },
  { key: 'mockEvent2',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:46:32.009Z',
      date: '2018-05-20',
      entryFee: '6',
      format: 'Standard',
      name: 'Tapahtuma',
      notes: 'FIJoijgasdsdgasdggagads',
      playerSlots: '14',
      prizes: 'Foooooooooooooooo',
      rulesLevel: 'Competitive',
      time: '10:00',
    } },
  { key: 'mockEvent3',
    value: {
      category: 'category2',
      createDate: '2018-05-20T16:04:29.595Z',
      date: '2018-05-21',
      entryFee: '5',
      format: 'Modern',
      link: 'https://magic.wizards.com/en/game-info/gameplay/formats/modern',
      name: 'Monday Modern',
      notes: 'Bring a friend!',
      playerSlots: '31',
      prizes: 'Three (3) OMG credits per player into the prizepool.',
      published: true,
      rulesLevel: 'Regular',
      time: '19:00',
    } },
  { key: 'mockEvent4',
    value: {
      category: '-LDIw-1LFgiM0EJZGfus',
      createDate: '2018-05-24T20:58:06.938Z',
      date: '2018-05-24',
      entryFee: '5',
      format: 'League',
      link: 'jokulinkki.fi',
      name: 'Necromunda League',
      notes: 'Shame! Shame! Shame! *DING DING DING*',
      playerSlots: '12',
      prizes: 'I am Loot.',
      published: true,
      rulesLevel: 'Casual',
      time: '17:00',
    } },
  { key: 'mockEvent5',
    value: {
      category: '-LAlvCmIWNzYKpa7-7HI',
      createDate: '2018-05-24T17:01:53.909Z',
      date: '2018-05-26',
      entryFee: '0',
      format: 'Board game',
      link: 'https://geekandsundry.com/international-tabletop-day-april-28-2018/',
      name: 'Lautapeli Lauantai',
      notes: 'Bring a friend and get a surprise gift!',
      playerSlots: '30',
      prizes: "A Supi's surprise prize for all participants!",
      published: true,
      rulesLevel: 'Casual',
      time: '00:00',
    } },
  { key: 'mockEventId1',
    value: {
      category: 'category1',
      createDate: '2018-05-24T17:00:46.500Z',
      date: '2018-09-24',
      entryFee: '10',
      format: 'Miniatures',
      link: 'http://www.google.fi',
      name: 'Warhammer thing',
      notes: 'Bring your war face!',
      playerSlots: '10',
      prizes: 'Glory!',
      published: true,
      rulesLevel: 'Casual',
      time: 'Kill everything',
    } },
];

export const mockSettings: Settings = {
  activeLogo: 'http://foo.com/image.png',
  dateFormat: 'DD.MM.YYYY',
  pageSubtitle: '',
  pageTitle: '',
  subtitleTextColor: '#000',
  theme: 'darkly',
  titleBarAngle: '150',
  titleBarColor: '#000',
  titleBarColor2: '#ec1c22',
  titleBarPercentage: '50',
  titleTextColor: '',
};

export const mockUploadedLogos: {[key: string]: UploadedFile} =
  {
    15271953483568546: { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152719372155055672': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152719272536668394': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '1527227370219137123': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152722965771313789': { downloadURL: 'http://foo.com/image.png', name: 'image' },
    '152722747904159438': { downloadURL: 'http://foo.com/image.png', name: 'image' },
  };

export const mockParticipations: {[key: string]: Participation} = {
  mockEventId1: {
    mockUserId: {
      comment: 'Jee',
      date: '2018-06-23T09:40:26.992Z',
      firstName: 'Test',
      lastName: 'Person',
      userId: 'mockUserId',
    },
  },
};

export const mockCategories: {[key: string]: Category} = {
  category1: {
    abbreviation: 'WH40K',
    logo: '1527227370219137123',
    name: 'Warhammer 40,000',
    type: 'Miniatures',
  },
  category2: {
    abbreviation: 'MtG',
    logo: '152722965771313789',
    name: 'Magic: The Gathering',
    type: 'TCG',
  },
  '-LAlvCmIWNzYKpa7-7HI': {
    abbreviation: 'Lautapelit',
    logo: '152719272536668394',
    name: 'Lautapelit / Board games',
    type: 'Boardgame',
  },
  '-LCmESny8m9JVEFQ_qOt': {
    abbreviation: 'PKM',
    logo: '152722747904159438',
    name: 'Pok√©mon',
    type: 'TCG',
  },
  '-LDImTAGxkzrorL0tQdi': {
    abbreviation: 'VG',
    logo: '152719372155055672',
    name: 'Videogames',
    type: 'Videogames',
  },
  '-LDIw-1LFgiM0EJZGfus': {
    abbreviation: 'Necromunda',
    logo: '15271953483568546',
    name: 'Necromunda',
    type: 'Miniatures',
  },
};
