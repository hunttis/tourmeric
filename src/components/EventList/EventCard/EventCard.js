import React, { Component, Fragment } from 'react';
import { Translate } from 'react-localize-redux';
import PropTypes from 'prop-types';
import moment from 'moment';
import _ from 'lodash';
import { isLoaded } from 'react-redux-firebase';

import { participantCount, checkParticipation } from '../../../api/eventApi';
import EventModal from '../EventModal-container';
import { CardFooterMobile } from './CardFooterMobile';
import { CardFooterDesktop } from './CardFooterDesktop';

export default class EventCard extends Component {

  state = { modalOpen: false };

  openModal() {
    this.setState({ modalOpen: true });
  }

  closeModal() {
    this.setState({ modalOpen: false });
  }

  render() {
    const { eventId, events, userId, participations, categories } = this.props;

    const eventContent = isLoaded(events) ? _.find(events, ['key', eventId]).value : {};
    const category = isLoaded(categories) ? categories[eventContent.category] : {};
    const alreadyParticipated = checkParticipation(userId, eventId, participations);
    const thisParticipation = _.get(participations, `${eventId}.${userId}`, {});
    const maxParticipants = _.get(eventContent, 'playerSlots', 0);
    const currentParticipants = participantCount(eventId, participations);
    const eventFull = Boolean(maxParticipants && maxParticipants <= currentParticipants);

    return (
      <Fragment>
        {this.state.modalOpen &&
          <EventModal key={`modal${eventId}`} eventId={eventId} closeModal={() => this.closeModal()} />
        }
        <div className="column is-12 eventcard">
          <div className="card card-shadow">

            <div className="card-header eventheader">
              <div className="card-header-icon eventheader-text tooltip" data-tooltip={category.name}>
                <img className="image is-64x64" src={category.image} alt="" />
                {category.abbreviation}:
              </div>
              <div className="card-header-title eventheader-text">
                <div>{eventContent.name}</div>
              </div>
            </div>

            <div className="card-content">
              <div className="has-icons-left card-item">
                <span className="tooltip" data-tooltip="Starting time">
                  <i className="fas fa-clock" />&nbsp;&nbsp;{eventContent.time}
                </span>
              </div>
              <div className="has-icons-left card-item">
                <span className="tooltip" data-tooltip="Entry fee">
                  <i className="fas fa-money-bill-alt" />&nbsp;&nbsp;{eventContent.entryFee}&nbsp;â‚¬
                </span>
              </div>

              <div className={`has-icons-left card-item ${eventFull && 'has-text-warning'}`}>
                <span className="tooltip" data-tooltip="People already enrolled">
                  <i className="fas fa-users" />&nbsp;&nbsp;{currentParticipants}&nbsp;
                  {eventContent.playerSlots &&
                  <span>/&nbsp;{eventContent.playerSlots}&nbsp;
                    {eventFull && <span>(<Translate id="eventfull" />)</span>}
                  </span>}
                </span>
              </div>

              <div className="has-icons-left card-item">
                <div className="level">
                  {eventContent.format &&
                    <div className="level-left card-footer-items">
                      <span className="tooltip card-margin-right" data-tooltip="Format">
                        <i className="fas fa-book" />&nbsp;&nbsp;{eventContent.format}&nbsp;
                      </span>
                    </div>
                  }

                  {eventContent.rules &&
                    <div className="level-item card-footer-items">
                      <span className="tooltip card-margin-right" data-tooltip="Rules Level">
                        <i className="fas fa-balance-scale" />&nbsp;&nbsp;{eventContent.rulesLevel}
                      </span>
                    </div>
                  }

                  <div className="level-right card-footer-items ">
                    <a onClick={() => this.props.history.push(`/event/${eventId}`)} className="card-footer-link">
                      <i className="fas fa-trophy" />&nbsp;&nbsp;<Translate id="allinfo" />
                    </a>
                  </div>

                </div>
              </div>

            </div>

            <CardFooterDesktop
              alreadyParticipated={alreadyParticipated}
              thisParticipation={thisParticipation}
              eventId={eventId}
              userId={userId}
            />

            <CardFooterMobile
              alreadyParticipated={alreadyParticipated}
              thisParticipation={thisParticipation}
              eventId={eventId}
              userId={userId}
            />


          </div>
          <p>&nbsp;</p>
          { moment(eventContent.date).isSameOrAfter(moment(), 'day') &&
          <div className="level participation-level">

            { !alreadyParticipated &&
              <div className="level-left is-hidden-mobile" />
            }

          </div>
        }
        </div>
      </Fragment>
    );
  }
}

EventCard.propTypes = {
  eventId: PropTypes.string.isRequired,
  userId: PropTypes.string,
  events: PropTypes.array.isRequired,
  participations: PropTypes.object,
  categories: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
};
