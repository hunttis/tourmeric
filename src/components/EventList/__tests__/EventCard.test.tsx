import React from 'react';
import { shallow } from 'enzyme';
import { createBrowserHistory } from 'history';
import EventCard from '../EventCard/EventCard';
import { mockEventId, mockUserId, mockProfile, mockEvents, mockSettings, mockUploadedLogos, mockParticipations, mockCategories } from '../__mocks__/mockData';

describe('EventCard Tests', () => {

  it('Compares EventCard to snapshot', () => {

    const history = createBrowserHistory();

    const mockEventsAsObject = {
      [mockEvents[0].key]: { ...mockEvents[0].value },
      [mockEvents[1].key]: { ...mockEvents[1].value },
    };

    const card = shallow(<EventCard
      eventId={mockEventId}
      userId={mockUserId}
      profile={mockProfile}
      events={mockEventsAsObject}
      eventsongoing={mockEventsAsObject}
      settings={mockSettings}
      participations={mockParticipations}
      categories={mockCategories}
      openModal={() => {}}
      uploadedCategoryLogos={mockUploadedLogos}
      history={history}
      setReturnLocation={() => {}}
    />);

    expect(card.exists()).toBe(true);

    const eventHeader = card.find('.eventheader');
    expect(eventHeader.text()).toBe('WH40K: Warhammer thing');
    expect(card).toMatchSnapshot();
  });

});
