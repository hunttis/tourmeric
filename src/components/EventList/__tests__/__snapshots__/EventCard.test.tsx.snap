// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard Tests Compares EventCard to snapshot 1`] = `
<Fragment>
  <div
    className="column is-12 eventcard"
  >
    <div
      className="card card-shadow "
    >
      <div
        className="card-content card-title card-logo-title "
      >
        <figure
          className="image event-icon-title event-item"
        >
          <img
            alt=""
          />
        </figure>
      </div>
      <div
        className="card-content"
      >
        <div
          className="media has-icons-right"
        >
          <div
            className="media-content event-card-title"
          >
            <p
              className="subtitle has-text-info"
            >
              20.05.2018 (Sunday)
            </p>
            <p
              className="title eventheader"
            >
              MtG
              : 
              Tapahtuma
            </p>
          </div>
        </div>
      </div>
      <div
        className="card-content"
      >
        <table
          className="table eventinfo-table"
        >
          <tbody>
            <CardInfoLine
              content="10:00"
              icon="fas fa-clock"
              title="startingtime"
            />
            <CardInfoLine
              content="6"
              icon="fas fa-money-bill-alt"
              title="entryfee"
            />
            <CardInfoLine
              content="0 / 14 "
              extraClasses=""
              icon="fas fa-users"
              title="participants"
            />
            <CardInfoLine
              content="Standard"
              icon="fas fa-book"
              title="format"
            />
          </tbody>
        </table>
        <div
          className="allinfolink"
        >
          <a
            className="card-footer-link"
            onClick={[Function]}
          >
            <i
              className="fas fa-trophy"
            />
              
            <Translate
              id="allinfo"
            />
          </a>
        </div>
      </div>
      <Component
        alreadyParticipated={false}
        eventId="mockEvent2"
        thisParticipation={Object {}}
        userId="mockUserId"
      />
      <Component
        alreadyParticipated={false}
        eventId="mockEvent2"
        thisParticipation={Object {}}
        userId="mockUserId"
      />
    </div>
    <p>
       
    </p>
  </div>
</Fragment>
`;
